post-process.py

```diff
@@ -0,0 +1,123 @@
+import os
+import pandas as pd
+from io import StringIO as sio
+import re
+import sys
+import argparse
+from pathlib import Path
+import glob
+
+""" Global Variables """
+
+file_name = "vernier-output-"
+
+""" Main funcs """
+
+def parse_cli_arguments(arguments: list[str] = None):
+
+    """ Parses command line arguments for running the script in the terminal """
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-path",          type=Path,  default=(os.getcwd()),                help="Path to Vernier output files")
+    parser.add_argument("-outputname",    type=str,   default=str("vernier-merged-output"), help="Name of file to write to.")
+
+    return parser.parse_args(args=arguments)
+
+def read_mpi_ranks(current_dir_path):
+
+    """ Reads the number of vernier-output files from the given directory to determine no. of ranks to use """
+
+    files = glob.glob(f"{current_dir_path}/{file_name}*")
+
+    return len(files)
+
+def read_and_pre_process(file_path, rank):
+
+    """ Reads in the current file """
+
+    file = open(f'{file_path}/{file_name}{rank}')
+    content = file.read()
```

Comment:
Can we have consistent spacing between the comments and the code they are describing, no space would be better.